tass/actions/selenium.py

```diff
@@ -68,3 +70,31 @@ def switch_frame(driver, find=_find_element, frame=None):
             driver.switch_to.frame(frame)
         else:
             driver.switch_to.frame(find(driver, **frame))
+
+
+# / / / / / / / Assertions / / / / / / /
+def assert_displayed(driver, find=_find_element, soft=False, **kwargs):

```

Comment:
I am now wondering if we should add an "expected" argument here. We could have it default to True, but for the odd circumstance where we want to check that an element is _not_ displayed it would be better to have it within a single function instead of having a second function to test for the opposite?

Reply:
@ty-desj is this a real assertion (throws exception or aborts if it fails)? If so, then we will need a different function, imo. It seems wrong to me to have one assertion that checks for both presence and absence by flipping a switch. You could have a `is_displayed()` function that is called in both assertions though. Returns true if displayed, and you just not it for the `assert_not_displayed()`.

If the assertion isn't a real one, then I would rename it (`is_displayed()` maybe?), and let the user deal with the issue. It's trivial to do something like `if(!is_displayed())` when you need to make sure something isn't present.

Reply:
@DominicParent there are 2 situations where I see this being used, the first being as a normal assertion: if the element is not displayed (or is displayed as the case may be) then an exception is thrown, the error is recorded for reporting, and the rest of the case is aborted.

The second situation would be more of a soft assertion. For example, in the case where the display status of an element does not prevent the case from continuing (maybe checking if a particular image, or logo is displayed). In this situation an exception would still be thrown, but the handling would be slightly different. The error is still recorded, but the test case can continue as normal.   

> then we will need a different function, imo. It seems wrong to me to have one assertion that checks for both presence and absence by flipping a switch.

My thoughts were to basically be something like 'assertEqual' thinking it might be more user friendly, but you are right, it does seem off somehow to have a binary success switch for this type of function. So then, something like this (but actual good code):

```
def assertDisplayed():
    if (!isDisplayed()):
         raise ...
```

```
def assertNotDisplayed():
    if (isDisplayed()):
         raise ...
```

Reply:
@ty-desj I like it!

Let me know when the PR is ready to be reviewed.

Reply:
Should be good to go now! @DominicParent